src/iface/ethernet.rs

```diff
@@ -990,4 +1011,56 @@ mod test {
             &IpAddress::Ipv4(remote_ip_addr)),
             Ok((remote_hw_addr, MockTxToken)));
     }
+
+    #[test]
+    #[cfg(feature = "proto-icmpv4")]
+    fn test_icmpv4_socket() {
+        use socket::{Icmpv4PacketBuffer, Icmpv4Socket, Icmpv4SocketBuffer};
+        use wire::{Icmpv4Repr, Icmpv4Packet};
+
+        let (iface, mut socket_set) = create_loopback();
+
+        let rx_buffer = Icmpv4SocketBuffer::new(vec![Icmpv4PacketBuffer::new(vec![0; 20])]);
+        let tx_buffer = Icmpv4SocketBuffer::new(vec![Icmpv4PacketBuffer::new(vec![0; 20])]);
+
+        let mut icmp_bytes = vec![0u8; 20];
+        let mut icmp_packet = Icmpv4Packet::new(&mut icmp_bytes);
+
+        let icmpv4_socket = Icmpv4Socket::new(rx_buffer, tx_buffer);
+
+        let socket_handle = socket_set.add(icmpv4_socket);
+
+        {
+            let mut icmp_socket = socket_set.get::<Icmpv4Socket>(socket_handle);
+
+            assert_eq!(icmp_socket.recv(Ipv4Address([0x7f, 0x00, 0x00, 0x02])),
+                       Err(Error::Exhausted));
+        }
+
+        let icmp_repr = Icmpv4Repr::TimestampRequest {
```

Comment:
That request is deprecated, no? I try to avoid adding deprecated things to `wire` unless they're used in practice.

Reply:
Yeah, it's not really used. It's really only used when echo requests are blocked by a firewall. I'll just manually write out the bytes for the test in the interface tests.

Reply:
Why not use an echo request?

Reply:
Updated